components/director/internal/domain/apptemplate/service.go

```diff
@@ -338,9 +338,61 @@ func (s *service) PrepareApplicationCreateInputJSON(appTemplate *model.Applicati
 
 		appCreateInputJSON = strings.ReplaceAll(appCreateInputJSON, fmt.Sprintf("{{%s}}", placeholder.Name), newValue)
 	}
+
+	for _, placeholder := range appTemplate.Placeholders {
+		newValue, err := values.FindPlaceholderValue(placeholder.Name)
+		isOptional := false
+		if placeholder.Optional != nil {
+			isOptional = *placeholder.Optional
+		}
+
+		if err != nil && !isOptional {
+			return "", errors.Wrap(err, "required placeholder not provided")
```

Comment:
add unit test for this case

Reply:
This statement is never reached, as the same check was done several lines above. 
Will look for a way to remove this redundant check.